core/chains/evm/client/client.go

```diff
@@ -54,6 +58,8 @@ type Client interface {
 	HeadByHash(ctx context.Context, n common.Hash) (*evmtypes.Head, error)
 	SubscribeNewHead(ctx context.Context, ch chan<- *evmtypes.Head) (ethereum.Subscription, error)
 
+	SendTransactionAndReturnCode(ctx context.Context, tx *types.Transaction, fromAddress common.Address) (clienttypes.SendTxReturnCode, error)
```

Comment:
What does `And` mean here? Does this just distinguish from the pre-existing `SendTransaction` method? If so, I think it would be fine to omit it. There is precedent in the std lib for introducing `*Context()` variants of functions, for example: https://pkg.go.dev/os/exec#Command & https://pkg.go.dev/os/exec#Command
> CommandContext is like Command but includes a context.
Would it be correct to add a similar comment here?
```suggestion
	// SendTransactionReturnCode is like SendTransaction but returns a [clienttypes.SendTxReturnCode].
	SendTransactionReturnCode(ctx context.Context, tx *types.Transaction, fromAddress common.Address) (clienttypes.SendTxReturnCode, error)
```

Reply:
`SendTransactionAndReturnCode` is not exactly the same as `SendTransaction`. It was named like that to indicate that the function does two things: sends a transaction, but also parses the result and returns the error code. In the example you provided, only the arguments differ not the return type, so the user expects the same output. I wouldn't say it's exactly the same case here.

Reply:
`SendTransactionAndReturnCode` is not exactly the same as `SendTransaction`. It was named like that to indicate that the function does two things: sends a transaction, but also parses the result and returns the error code. In the example you provided, only the arguments differ not the return type, so the user expects the same output. I wouldn't say it's exactly the same case here.

Reply:
Could we just delete the old SendTransaction, and rename the current one as SendTransaction?
I see there's no other caller for it, except the txmgr

Reply:
Ah, as in `(send transaction) and (return code)`, not `(send transaction) and (type ReturnCode)`?
On my first read, it looked like `(send (transaction and return code))`. Can we try to find a better name?
`SendTransactionReturnCode` still seems fitting to me, unless I'm misunderstanding something. It is not exactly the same as the linked example, but the more general pattern is to suffix the name with the newly added type.

Reply:
Alternatively, what about `SendTransactionFrom` since `fromAddress` is also new, and we don't even mention the return code, we just document it?
```suggestion
	SendTransactionFrom(ctx context.Context, tx *types.Transaction, fromAddress common.Address) (clienttypes.SendTxReturnCode, error)
```